api/src/pcapi/core/providers/api.py

```diff
@@ -378,7 +378,11 @@ def _should_reindex_offer(new_quantity: int, new_price: float, existing_stock: d
     if existing_stock["price"] != new_price:
         return True
 
-    is_existing_stock_empty = existing_stock["quantity"] <= existing_stock["booking_quantity"]
+    is_existing_stock_empty = (
+        existing_stock["quantity"] <= existing_stock["booking_quantity"]
+        if existing_stock["quantity"] is not None  # stock quantity infinite
+        else False
+    )
```

Comment:
(style / question de goût) variante :

```suggestion
    is_existing_stock_empty = (
        # Existing stock could be None (i.e. infinite) if the offerer manually overrode
        # the quantity of this synchronized stock.
        existing_stock["quantity"] is not None
        and existing_stock["quantity"] <= existing_stock["booking_quantity"]
    )
```